code/modules/antagonists/changeling/headslug_eggs.dm

```diff
@@ -0,0 +1,44 @@
+#define EGG_INCUBATION_TIME (4 MINUTES)
+
+/// The actual organ that the changeling headslug implants into a dead host.
+/obj/item/organ/internal/body_egg/changeling_egg
+	name = "changeling egg"
+	desc = "Twitching and disgusting."
+	/// The mind of the original changeling that gave forth to the headslug mob.
+	var/datum/mind/origin
+	/// Tracks how long the egg has been growing.
+	var/time = 0
+
+/obj/item/organ/internal/body_egg/changeling_egg/egg_process(seconds_per_tick, times_fired)
+	// Changeling eggs grow in dead people
+	time += seconds_per_tick * 10
```

Comment:
Why can't this use cooldowns instead?

Reply:
unfortunately this would require a huge overhaul of the whole body_egg framework which is out of scope for the intents and purposes of this PR